addons/pos_self_order_sale/models/res_config_settings.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, api
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    @api.onchange("pos_self_order_kiosk")
+    def _self_order_kiosk_change(self):
+        for record in self:
+            if record.pos_config_id.self_order_kiosk:
+                record.pos_config_id.self_order_view_mode = False
+                record.pos_config_id.self_order_table_mode = False
```

Comment:
It is already in res_config_settings.py in pos_self_order module

Reply:
?